sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -695,16 +692,14 @@ public virtual async Task<Response<RecognizeEntitiesResultCollection>> Recognize
         /// status code.</exception>
         public virtual Response<RecognizeEntitiesResultCollection> RecognizeEntitiesBatch(IEnumerable<TextDocumentInput> documents, TextAnalyticsRequestOptions options = default, CancellationToken cancellationToken = default)
         {
-            //Argument.AssertNotNullOrEmpty(documents, nameof(documents));
-            //options ??= new TextAnalyticsRequestOptions();
-            ////MultiLanguageBatchInput documentInputs = ConvertToMultiLanguageInputs(documents);
+            Argument.AssertNotNullOrEmpty(documents, nameof(documents));
+            options ??= _requestOptions;
+            AnalyzeTextEntityRecognitionInput input = new AnalyzeTextEntityRecognitionInput();
+            input = TextDocumentInputToEntityRecognition(documents, input, options);
 
-            //AnalyzeTextEntityRecognitionInput input = ConvertToMultiLanguageInputs(documents);
-            //return RecognizeEntitiesBatch(input, options, cancellationToken);
-            throw new NotImplementedException();
+            return RecognizeEntitiesBatch(input, options, cancellationToken);
         }
 
-        //CHANGED: Overload for new Swagger
         private async Task<Response<RecognizeEntitiesResultCollection>> RecognizeEntitiesBatchAsync(AnalyzeTextEntityRecognitionInput entityRecognitionInput, TextAnalyticsRequestOptions options, CancellationToken cancellationToken)
```

Comment:
`entityRecognitionInput` was already populated in the `TextDocumentInputToEntityRecognition`. There is no need to create a new `AnalyzeTextEntityRecognitionInput` and assign all the values again

Reply:
this has not been resolved, and this is actually adding a bug because `StringIndexType` is never set.
Let me know if you have questions about my comment

Reply:
I think I set it in line 768. StringIndexType is the last parameter

Reply:
Correct, but look at the flow of this function:
- You get as a parameter `AnalyzeTextEntityRecognitionInput entityRecognitionInput` which indeed was correctly set.
- In line 705  you create a new `AnalyzeTextEntityRecognitionInput` called `analyzeRecognizeEntities ` and assign to it `entityRecognitionInput.AnalysisInput`
- line 706 and 707 you assign to `analyzeRecognizeEntities` parameters that don't include `StringIndexType`
- line 710 you are passing to the `AnalyzeAsync` method the `analyzeRecognizeEntities` that you created in this function

So in the end, you are not passing the pretty `entityRecognitionInput` that you created before.

My point is, you already did all the work to correctly populate `entityRecognitionInput`, why not use that one directly for the `AnalyzeAsync` method?

Reply:
sorry, I understood it now. updated

Reply:
I was using a lot of extra memory before